src/main/java/com/amam/wizardschool/service/StudentService.java

```diff
@@ -119,4 +120,47 @@ public double getAverageAgeByStream() throws StudentNotFoundException {
                 .average()
                 .orElseThrow(() -> new StudentNotFoundException("No students have found"));
     }
+
+    public void printParallel() {
+        List<Student> students = studentRepository.findAll().subList(0, 6);
+
+        printNames(students.subList(0, 2));
+        new Thread(() -> printNames(students.subList(2, 4))).start();
+        new Thread(() -> printNames(students.subList(4, 6))).start();
+    }
+
+    public void printSynchronized() throws InterruptedException {
+        List<Student> students = studentRepository.findAll().subList(0, 6);
+
+        printNamesSync(students.subList(0, 2));
+        new Thread(() -> {
+            try {
+                printNamesSync(students.subList(2, 4));
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }).start();
+
+        new Thread(() -> {
+            try {
+                printNamesSync(students.subList(4, 6));
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }).start();
+
+    }
+
+    private void printNames(List<Student> students) {
+        for (int i = 0; i < 2; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < students.size(); i++) {
```
размер списка надо использовать, а не число с потолка